pkgs/development/tools/manifest-tool/default.nix

```diff
@@ -29,14 +28,23 @@ buildGoModule rec {
 
   nativeBuildInputs = [ git ];
 
-  preConfigure = ''
-    ldflags="-X main.gitCommit=$(cat .git-revision)"
-  '';
-
-  CGO_ENABLED = if stdenv.hostPlatform.isStatic then "0" else "1";
-  GO_EXTLINK_ENABLED = if stdenv.hostPlatform.isStatic then "0" else "1";
-  ldflags = lib.optionals stdenv.hostPlatform.isStatic [ "-w" "-extldflags" "-static" ];
-  tags = lib.optionals stdenv.hostPlatform.isStatic [ "netgo" ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+    "-X main.gitCommit=${builtins.readFile "${src}/.git-revision"}"
```

Comment:
Looks like an IFD?

Reply:
@wegank I am not really familiar with the topic of Import From Derivation (IFD).

Can you help me😃? I want to read a version file which I generated during the postFetch phase. Before I read this file with a `cat` command in the preConfigurePhase.

So is this now an IFD? Are they considered bad?

Reply:
@infinisil Could me help me understand if the above is an IFD😃?

Reply:
IFD is disallowed in Nixpkgs, [see here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#import-from-derivation).

As you said, just do `ldflags+="-X main.gitCommit=$(cat .git-revision)"` in `preConfigure` instead.

Reply:
@wegank Thanks for the info.  I removed the IFD again as you suggested.